src/go/Rumble/rumble_server.go

```diff
@@ -0,0 +1,394 @@
+        err := cmd.Run()
+        responseErrMsg.SysError = fmt.Sprintf("%v", err)
+        responseErrMsg.Stdout = out.String()
+        appId := parseAppId(out.String())
+        log.Println("[INFO] Stdout:", out.String())
+        log.Println("[DEBUG] Stderr:", stderr.String())
+        if verbose > 1 { // If user sets verbose greater than 1, send all errors.
+            responseErrMsg.VerboseMsg = stderr.String()
+        }
+        if err != nil {
+            responseErrMsg.Message = "Rumble execution failed. Please check your query! "
+            log.Printf("[ERROR] Spark-submit failed: %v", responseErrMsg)
+            return Response{
+                ErrorMsg:   responseErrMsg,
+                SparkAppId: appId,
+            }, false
+        }
+        if verbose > 0 {
+            responseErrMsg.Message = "Rumble execution SUCCESSFUL. You can reach your data from: " + outputPath
+            return Response{
+                UserId:            userId,
+                SparkAppId:        appId,
+                ErrorMsg:          responseErrMsg,
+                OutputPath:        outputPath,
+                InternalQueryFile: queryFile,
+            }, true
+        }
+        return Response{
+            UserId:     userId,
+            SparkAppId: appId,
+            OutputPath: outputPath,
+            ErrorMsg: ErrorMessage{
+                Message: "Rumble execution SUCCESSFUL. You can reach your data from: " + outputPath,
+            },
+        }, true
+    }
+}
+
+func parseAppId(out string) string {
+    flagStart, flagEnd := "ApplicationID:[", "]APP_ID_FLAG"
+    if strings.Contains(out, flagStart) {
+        return strings.Split(strings.Split(out, flagStart)[1], flagEnd)[0]
+    }
+    return ""
+}
+
+func sanitizeOutputPath(file string) string {
+    checkStr := []string{"hdfs:///analytix", "hdfs://analytix", "hdfs://", "hdfs:/", "hdfs:", "hdfs"}
+    for _, substr := range checkStr {
+        if strings.Contains(file, substr) {
+            file = strings.Replace(file, substr, "", 1)
+            break
+        }
+    }
+    if !strings.HasPrefix(file, "/") {
+        return "hdfs://analytix/" + file
+    }
+    return "hdfs://analytix" + file
+}
+
+// writeQueryToFile Write JSONiq query to a file with unix timestamp.
+func writeQueryToFile(query string, userId string) (string, error) {
+    baseQueryPath := "/data/queries/"
+    _ = os.MkdirAll(baseQueryPath, os.ModeDir)   // Create if it does not exist, else do nothing
+    userId = strings.ReplaceAll(userId, " ", "") // Clear whitespace
+    jsoniqFileName := fmt.Sprintf("%s%s-%d.jq", baseQueryPath, userId, time.Now().UnixNano())
+    if err := ioutil.WriteFile(jsoniqFileName, []byte(query), 0666); err != nil {
+        log.Println("[ERROR] could not create temp file, reason:", err)
+        return "", err
+    }
+    return jsoniqFileName, nil
+}
+
+func writeMapToFile(portsInUse map[string]bool, lockFname string) {
+    data, _ := json.Marshal(portsInUse)
```

Comment:
please do not skip errors, json.Marshal can fail too. Check the error as well as it will be lost otherwise and debugging issue will be very tedious process.